### DISCUSSION TITLE
external TCP / stationary tool

### DISCUSSION THREAD

**Discussion by alekskarai** [visose/Robots#157] [Q&A] [answered]
Hello,

I was wondering if its possible to define a stationary tool and a moving workpiece.
The transformations can be computed without this option, treating the workpiece like a tool, but it is hard to control the speed at the actual (static) TCP.
The application is a hot wire cutter with the robot (KUKA KRC4) holding the workpiece to be cut.

Thanks,
Alex

**Comment by visose** [comment, marked as answer]
> I was wondering if its possible to define a stationary tool and a moving workpiece.

Yes, you can do it as you mention, using a frame parameter to define the location of the static tool. This is usually referred to "part-to-tool".

The speed should be the same irrespective of if the part moves or the tool moves.
<img width="241" alt="relative" src="https://user-images.githubusercontent.com/5129614/165907542-2801af57-37cf-4beb-b0d3-c8408331d14d.png">

**Reply by kasf21** [reply]
Hello,

I'm sorry, I don't understand what you mean by the above.
Do you have any examples of the above?
